### ISSUE TITLE
Columnar Metadata

### ISSUE THREAD

**Issue by jpwkeeper** [G-Research/ParquetSharp#148]
I cannot find a way to add custom columnar metadata.  I can read it, but I can't figure out how to write it.  Am I missing something?

**Comment by GPSnoopy** [comment]
Hi @jpwkeeper, do you have an example of metadata (i.e. code) that you can read but not write?

**Comment by jpwkeeper** [comment]
You know, I swore I could, but now that I look ColumnChuckMetaData doesn't seem to support custom metadata either.

**Comment by GPSnoopy** [comment]
Yes, as far as I'm aware, you can only add custom key-value pairs on a per-file basis. I was wondering if we missed any API from Arrow/Parquet that allows for more fine grained metadata.

**Comment by jpwkeeper** [comment]
Maybe I mis-understood some of what I read about Parquet.  One of the struggles of being a .NET developer is that it's hard to find any examples or tutorials on this type of thing, and the examples for other languages like Scala are extremely difficult to understand if you don't know the language.
